src/null-safety/index.md

```diff
@@ -1,39 +1,247 @@
+{:.table .table-striped}
+
+When literals initialize a list or set,
+then instead of seeing a type like in the table above,
+you typically see a type annotation on the literal.
+For example, here’s what a `List<String?>` (called `nameList`) and
+a `Set<String?>` (called `nameSet`) might look like:
+
+```dart
+var nameList = <String?>['Andrew', 'Anjan', 'Anya'];
+var nameSet = <String?>{'Andrew', 'Anjan', 'Anya'};
+```
+
+### Map types {#map-types}
+
+Map types behave mostly like you’d expect, with one exception: **the returned value can be null**. Null is the value for a key that isn't present in the map.
+
+As an example, look at the following code. What do you think the value and type of `uhOh` are?
+
+```dart
+var myMap = <String, int>{'one': 1};
+var uhOh = myMap['two'];
+```
+
+The answer is that `uhOh` is null and has type `int?`. 
+
+Like lists and sets, maps can have a variety of type permutations.
+
+|----------------------+-------------------------------+-------------------------|
+|Type                  | Can an item (int)<br>be null? | Can the map<br>be null? |
+|----------------------|-------------------------------|-------------------------|
+| `Map<String, int>`   | No*                           | No                      |
+| `Map<String, int>?`  | No*                           | {{yes}}                 |
+| `Map<String, int?>`  | {{yes}}                       | No                      |
+| `Map<String, int?>?` | {{yes}}                       | {{yes}}                 |
+{:.table .table-striped}
+
+_\* Even when all the int values in the map are non-null,
+when you use an invalid key to do a map lookup, the returned value is null._
+
+Because map lookups can return null,
+you can't assign them to non-nullable variables:
+
+```dart
+// Assigning a lookup result to a non-nullable
+// variable causes an error.
+int value = <String, int>{'one': 1}['one']; // ERROR
+```
+
+**[QUESTION: Will the analyzer ever be able to detect that is non-null?]**
+
+One workaround is to change the type of the variable to be nullable:
+
+```dart
+int? value = <String, int>{'one': 1}['one']; // OK
+```
+
+Another way to fix the problem —
+if you're sure the lookup succeeds —
+is to add a `!`:
+
+```dart
+int value = <String, int>{'one': 1}['one']!; // OK
+```
+
+A safer approach is to use the lookup value only if it's not null.
+You can test its value using
+an [`if` statement][] or the [`??` operator][].
+Here's an example of using the value `0` if the lookup returns a null value:
+
+```dart
+var aList = <String, int>{'one': 1};
+...
+int value = aList['one'] ?? 0;
+```
+
+[`if` statement]: /guides/language/language-tour#if-and-else
+[`??` operator]: /guides/language/language-tour#conditional-expressions
+
+
+## Fixing analysis errors
+
+With sound null safety, the Dart analyzer generates errors when
```

Comment:
Isn't "sound null safety" the phrase we use for the sound run-time mode.
Here it seems to be used just for "null safety", so maybe remove the "sound".